### PR TITLE
Add Dev Workflow Docs

### PR DESCRIPTION
**Pull request by willtebbutt**
Addresses #268 . @sunxd3 would you mind taking a quick look at this and letting me know if it is clear? I feel like you have most experience developing Mooncake after me, so you are likely best positioned to judge this.

### CODE REVIEW

File: docs/src/running_tests_locally.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+Mooncake's tests are organised as follows:
+1. Things that are required for most / all test suites are loaded up in `test/front_matter.jl`.
+1. The tests for something in `src` are located in an identically-named file in `test`. e.g. the unit tests for `src/rrules/new.jl` are located in `test/rrules/new.jl`.
+
+Thus, a workflow that I (Will) find works very well is the following:
+1. Ensure that you have Revise.jl and TestEnv.jl installed in your default environment.
+1. start the REPL, `dev` Mooncake.jl, and navigate to the top level of the Mooncake.jl directory.
+1. `using TestEnv, Revise`. Better still, load both of these in your `.julia/config/startup.jl` file so that you don't ever forget to load them.
+1. Run the following: `using Pkg; Pkg.activate("."); TestEnv.activate(); include("test/front_matter.jl");` to set up your environment.
```
Comment (sunxd3) [inline docs/src/running_tests_locally.md:13]: not a suggestion, just a anecdotal point from me: I configure VSCode to automatically do the `Pkg.activate()` for the package folder I open. But what you wrote is better as it assumes less.

**Comment by sunxd3** [comment]
All make sense, thanks Will. I use VSCode REPL integration quite often, particularly `ctrl/shift+enter` to execute code.

File: docs/src/running_tests_locally.md (hunk 1 above)
Comment (willtebbutt) [reply docs/src/running_tests_locally.md:13]: Thanks for letting me know -- this is good to know.

**Comment by willtebbutt** [comment]
Thanks for the review @sunxd3  -- I'll merge when CI passes.

**Pull request merged**